src/main/scala/com/databricks/spark/redshift/RedshiftJDBCWrapper.scala

```diff
@@ -141,7 +141,13 @@ private[redshift] class JDBCWrapper extends Logging {
         case ShortType => "INTEGER"
         case ByteType => "SMALLINT" // Redshift does not support the BYTE type.
         case BooleanType => "BOOLEAN"
-        case StringType => "TEXT"
+        case StringType =>
+          val maxlength = if (field.metadata.contains("maxlength")) {
+            field.metadata.getLong("maxlength")
+          } else {
+            255 // TODO: make this default configurable?
```

Comment:
Should we just use `TEXT` in this case, which essentially defers the default to redshift?